### ISSUE TITLE
argo version upgrade for nested argo parameter support

### ISSUE THREAD

**Issue by pzou1974** [keikoproj/addon-manager#26]
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE REQUEST?

**What happened**:
addon mgr does not support nested parameter due to argo does not support

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Addon Manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- Addon status:
```
$ kubectl describe addon <addon-name>
```
- controller logs:
```
$ kubectl logs <addon-manager-pod>
```

**Comment by pzou1974** [comment]
new:https://github.com/argoproj/argo/releases/tag/v2.4.1
This release contains Bug fixes from 2.4.0
1. Stop failing if artifact file exists, but empty (#1653) (@Ark-kun)
2. Don’t provision VM for empty artifacts(#1660) (@simster7)
3. Resolve WorkflowTemplate lazily (#1655) (@dtaniwaki )

**Comment by pzou1974** [comment]
https://raw.githubusercontent.com/argoproj/argo/v2.4.1/manifests/install.yaml
